eng/common/scripts/Package-Properties.ps1

```diff
@@ -90,14 +90,20 @@ class PackageProps {
                 $result = [PSCustomObject]@{
                     ArtifactConfig = [HashTable]$artifactForCurrentPackage
                     MatrixConfigs  = @()
+                    AdditionalMatrixConfigs = @()
                 }
 
                 # if we know this is the matrix for our file, we should now see if there is a custom matrix config for the package
                 $matrixConfigList = GetValueSafelyFrom-Yaml $content @("extends", "parameters", "MatrixConfigs")
-                # JRS - it's unclear if this is the place add AdditionalMatrixConfigs, it seems
-                # that if this is set, it'll replace the basic test matrix
+
                 if ($matrixConfigList) {
-                    $result.MatrixConfigs = $matrixConfigList
+                    $result.MatrixConfigs += $matrixConfigList
+                }
+
+                $additionalMatrixConfigList = GetValueSafelyFrom-Yaml $content @("extends", "parameters", "AdditionalMatrixConfigs")
+
+                if ($additionalMatrixConfigList) {
+                    $result.AdditionalMatrixConfigs += $additionalMatrixConfigList
```

Comment:
`GetValueSafelyFrom-Yaml` is victim to array unrolling, so our previous use case was accidentally changing type, and adding the additionalmatrix config to it was causing issues.

This PR fixes that + optionally parses the AdditionalMatrixConfigs param. We should start seeing the version_override generate correctly for `azure-core`.

Reply:
Locally at least I see it generating properly.

Reply:
@scbedd this works. If you look at the PR [pipeline run](url) there are 9 test runs from the default matrix and 13 version override additional matrix config. This matches what [java - core runs](https://dev.azure.com/azure-sdk/public/_build/results?buildId=4538094&view=results). I'll get these changes into an eng/common PR. Thank you.